### DISCUSSION TITLE
python version 3.10.13 in the docker container

### DISCUSSION THREAD

**Discussion by PierreLouis94** [Kitware/trame#622] [Trame/ParaView]
Hello, i was scanning my docker container with the vuln scanner grype. One of the vulnerabilities detected with High severtiy was the python package 3.10.13.

I first scanned my base image which is kitware/trame:py3.10-glvnd-2024-10
here i couddnt find this vulnerability. So the vulnerability must have been added further down my dockerfile. 

```
FROM kitware/trame:py3.10-glvnd-2024-10

# Update package lists and upgrade libc-bin to the latest version
RUN apt-get update && apt-get install -y libc-bin && apt-get upgrade -y libc-bin && apt-get clean
  
RUN install -d -o trame-user -g trame-user /deploy
  
ARG PV_URL='https://www.paraview.org/files/v5.13/ParaView-5.13.0-egl-MPI-Linux-Python3.10-x86_64.tar.gz'
RUN mkdir -p /opt/paraview && cd /opt/paraview && wget -qO- $PV_URL | tar --strip-components=1 -xzv
ENV TRAME_PARAVIEW=/opt/paraview
ENV TRAME_CLIENT_TYPE=vue2
  
COPY --chown=trame-user:trame-user . /deploy

  
RUN /opt/trame/entrypoint.sh build
```

I wanted to ask if this python version 3.10.13 is related with the introduction of the paraview-5.13.0 binary. Could it be?

**Comment by jourdain** [comment]
Either it could be the Python bundled in ParaView or the old cached version of `kitware/trame:py3.10-glvnd-2024-10` that you are building against?

**Reply by PierreLouis94** [reply]
Hi Sébastien,

Thanks for the quick response! I double-checked by scanning the base image (kitware/trame:py3.10-glvnd-2024-10) itself, and there was no sign of the Python 3.10.13 vulnerability there. This suggests that the vulnerability likely wasn’t introduced from the base image but may have come from the ParaView 5.13.0 binary.

**Reply by jourdain** [reply]
You may want to try PV 5.13.1?

**Reply by PierreLouis94** [reply]
Also tried, unfortunately the python version remains 3.10.13 in this version as well.

**Reply by jourdain** [reply]
What is the security issue with 3.10.13? I'm guessing the nightly built are also on that version of python?

**Reply by PierreLouis94** [reply]
These are the vulnerabilities categorized as High from the grype vulnerability scanner:
[CVE-2024-7592](https://nvd.nist.gov/vuln/detail/CVE-2024-7592), [CVE-2024-6232 ](https://nvd.nist.gov/vuln/detail/CVE-2024-6232 ), [CVE-2024-4032](https://nvd.nist.gov/vuln/detail/CVE-2024-4032), [CVE-2024-0397](https://nvd.nist.gov/vuln/detail/CVE-2024-0397), [CVE-2023-6597](https://nvd.nist.gov/vuln/detail/CVE-2023-6597), [CVE-2023-36632](https://nvd.nist.gov/vuln/detail/CVE-2023-36632)
A proposed fix to most them would be to update it from 3.10.13 to 3.10.15
The nightly build has also been tested and scanned, and it too is based on version 3.10.13.
I have also reviewed the [Paraview's dependency page ](https://www.paraview.org/files/dependencies/)  but could not find any Python version higher than 3.10.13.

**Reply by jourdain** [reply]
Thanks for those details, I reported those to the ParaView team. They might be able to update our CI to update the Python version used...

**Reply by PierreLouis94** [reply]
Thanks for following up with the ParaView team about this. I was wondering if there’s any update on when we might expect the Python version in the CI to be updated? Knowing a rough timeline would really help us plan on our end.

**Reply by jourdain** [reply]
I don't know when will get to it. But most likely for the next ParaView release in January.
